### PR TITLE
NetObserv 1.7 FlowCollector API regeneration

### PR DESCRIPTION
**Pull request by skrthomas**
Note to reviewers: This is autogenerated from https://github.com/netobserv/network-observability-operator/pull/796/files, which is still open and could accept some changes but some suggestions may be deferred until the next regeration of the API. 
Version(s):

Merge only to no-1.7. On GA 10/21/24, Sara will open a branch to integrate no-1.7 with main, and that branch will go to 4.12+.
Issue:

This PR spans multiple tickets, as it comprises API updates for the entire release. 
Link to docs preview:

https://83175--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/network_observability/flowcollector-api.html
QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 1]
```diff
@@ -2440,6 +2761,52 @@ Type::
 
 
 
+== .spec.processor.advanced.secondaryNetworks
+Description::
++
+--
+Define secondary networks to be checked for resources identification.
+In order to guarantee a correct identification, it is important that the indexed values form an unique identifier across the cluster.
+If there are collisions in the indexes (same index used by several resources), those resources might be wrongly labelled.
```
[the remark this answers is not in the document]
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:2770]: see https://github.com/netobserv/network-observability-operator/pull/796/files#r1797373805

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 2]
```diff
@@ -2440,6 +2761,52 @@ Type::
...
+
+[cols="1,1,1",options="header"]
+|===
+| Property | Type | Description
+
+| `index`
+| `array (string)`
+| `index` is a list of fields to use for indexing the pods. They should form a unique Pod identifier across the cluster.
+Can be any of: `MAC`, `IP`, `Interface`.
+Fields absent from the 'k8s.v1.cni.cncf.io/network-status' annotation must not be added to the index.
```
[the remark this answers is not in the document]
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:2803]: This is an annotation, so n/a

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 3]
```diff
@@ -2440,6 +2761,52 @@ Type::
...
+
+| `name`
+| `string`
+| `name` should match the network name as visible in the pods annotation 'k8s.v1.cni.cncf.io/network-status'.
```
[the remark this answers is not in the document]
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:2807]: n/a bc this is part of the annotation

**Comment by skrthomas** [comment]
@jotak @memodi I just need approvals here for merge. I commented a couple minor things on the regeneration PR. https://github.com/netobserv/network-observability-operator/pull/796/files#

**Comment by jotak** [comment]
Hi @skrthomas [I updated my PR](https://github.com/netobserv/network-observability-operator/pull/796) do you want to check before we merge this one?

**Review by memodi — APPROVED** [review]
/label qe-approved

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 4]
```diff
@@ -200,13 +205,18 @@ Otherwise it is matched as a case-sensitive string.
...
 - `PacketDrop`: enable the packets drop flows logging feature. This feature requires mounting
-the kernel debug filesystem, so the eBPF pod has to run as privileged.
+the kernel debug filesystem, so the eBPF agent pods have to run as privileged.
 If the `spec.agent.ebpf.privileged` parameter is not set, an error is reported. +
 
 - `DNSTracking`: enable the DNS tracking feature. +
 
 - `FlowRTT`: enable flow latency (sRTT) extraction in the eBPF agent from TCP traffic. +
 
+- `NetworkEvents`: enable the Network events monitoring feature, such as correlating flows and network policies.
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:215]:
```suggestion
- `NetworkEvents`: enable the network events monitoring feature, such as correlating flows and network policies.
```
Should "Network" be capitalized here?

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 5]
```diff
@@ -200,13 +205,18 @@ Otherwise it is matched as a case-sensitive string.
...
+This feature requires mounting the kernel debug filesystem, so the eBPF agent pods have to run as privileged.
+It requires using the OVN-Kubernetes network plugin with the Observability feature.
+IMPORTANT: this feature is available as a Developer Preview. +
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:218]:
```suggestion
IMPORTANT: This feature is available as a Developer Preview. +
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 6]
```diff
@@ -401,11 +412,16 @@ To filter a range of ports, use a "start-end" range in string format. For exampl
 | `peerIP` defines the IP address to filter flows by.
 Example: `10.10.10.10`.
 
+| `pktDrops`
+| `boolean`
+| `pktDrops`, to filter flows with packet drops
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:417]: Should line 417 be, "`pktDrops` filters flows with packet drops"?

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 7]
```diff
@@ -440,7 +461,7 @@ Type::
 | `disableAlerts` is a list of alerts that should be disabled.
 Possible values are: +
 
-`NetObservDroppedFlows`, which is triggered when the eBPF agent is dropping flows, such as when the BPF hashmap is full or the capacity limiter is being triggered. +
+`NetObservDroppedFlows`, which is triggered when the eBPF agent is missing packets or flows, such as when the BPF hashmap is busy or full, or the capacity limiter is being triggered. +
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:464]:
```suggestion
`NetObservDroppedFlows` is triggered when the eBPF agent is missing packets or flows, such as when the BPF hashmap is busy or full, or the capacity limiter is triggered. +
```
Should "BPF hashmap" be "eBPF hashmap"?

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 8]
```diff
@@ -1211,6 +1238,267 @@ Type::
...
+
+
+[cols="1,1,1",options="header"]
+|===
+| Property | Type | Description
+
+| `fieldsMapping`
+| `array`
+| Custom fields mapping to an OpenTelemetry conformant format.
+By default, Network Observability format proposal is used: https://github.com/rhobs/observability-data-model/blob/main/network-observability.md#format-proposal .
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1290]:
```suggestion
By default, Network Observability format proposal is used: https://github.com/rhobs/observability-data-model/blob/main/network-observability.md#format-proposal.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 9]
```diff
@@ -1211,6 +1238,267 @@ Type::
...
+As there is currently no accepted standard for L3 or L4 enriched network logs, you can freely override it with your own.
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1291]:
```suggestion
As there is currently no accepted standard for L3 or L4 enriched network logs, you can override it with your own.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 10]
```diff
@@ -1211,6 +1238,267 @@ Type::
...
+| `object`
+| OpenTelemetry configuration for metrics.
+
+| `protocol`
+| `string`
+| Protocol of the OpenTelemetry connection. The available options are `http` and `grpc`.
+
+| `targetHost`
+| `string`
+| Address of the OpenTelemetry receiver
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1311]:
```suggestion
| Address of the OpenTelemetry receiver.
```
Adding closing punctuation for consistency with other descriptions.

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 11]
```diff
@@ -1211,6 +1238,267 @@ Type::
...
+
+| `targetPort`
+| `integer`
+| Port for the OpenTelemetry receiver
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1315]:
```suggestion
| Port for the OpenTelemetry receiver.
```
Same as above.

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 12]
```diff
@@ -1211,6 +1238,267 @@ Type::
...
+| `object`
+| TLS client configuration.
+
+|===
+== .spec.exporters[].openTelemetry.fieldsMapping
+Description::
++
+--
+Custom fields mapping to an OpenTelemetry conformant format.
+By default, Network Observability format proposal is used: https://github.com/rhobs/observability-data-model/blob/main/network-observability.md#format-proposal .
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1327]:
```suggestion
By default, Network Observability format proposal is used: https://github.com/rhobs/observability-data-model/blob/main/network-observability.md#format-proposal.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 13]
```diff
@@ -1211,6 +1238,267 @@ Type::
...
+As there is currently no accepted standard for L3 or L4 enriched network logs, you can freely override it with your own.
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1328]:
```suggestion
As there is currently no accepted standard for L3 or L4 enriched network logs, you can override it with your own.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 14]
```diff
@@ -1211,6 +1238,267 @@ Type::
...
+|===
+| Property | Type | Description
+
+| `enable`
+| `boolean`
+| Set `enable` to `true` to send metrics to an OpenTelemetry receiver.
+
+| `pushTimeInterval`
+| `string`
+| How often should metrics be sent to collector
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1412]:
```suggestion
| Specify how often metrics are sent to a collector.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 15]
```diff
@@ -1211,6 +1238,267 @@ Type::
...
+
+
+
+[cols="1,1,1",options="header"]
+|===
+| Property | Type | Description
+
+| `caCert`
+| `object`
+| `caCert` defines the reference of the certificate for the Certificate Authority
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1434]:
```suggestion
| `caCert` defines the reference of the certificate for the Certificate Authority.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 16]
```diff
@@ -1211,6 +1238,267 @@ Type::
...
+| Enable TLS
+
+| `insecureSkipVerify`
+| `boolean`
+| `insecureSkipVerify` allows skipping client-side verification of the server certificate.
+If set to `true`, the `caCert` field is ignored.
+
+| `userCert`
+| `object`
+| `userCert` defines the user certificate reference and is used for mTLS (you can ignore it when using one-way TLS)
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1447]:
```suggestion
| `userCert` defines the user certificate reference and is used for mTLS. When you use one-way TLS, you can ignore this property.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 17]
```diff
@@ -1211,6 +1238,267 @@ Type::
...
+
+|===
+== .spec.exporters[].openTelemetry.tls.caCert
+Description::
++
+--
+`caCert` defines the reference of the certificate for the Certificate Authority
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1454]:
```suggestion
`caCert` defines the reference of the certificate for the Certificate Authority.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 18]
```diff
@@ -1211,6 +1238,267 @@ Type::
...
+| `type`
+| `string`
+| Type for the certificate reference: `configmap` or `secret`.
+
+|===
+== .spec.exporters[].openTelemetry.tls.userCert
+Description::
++
+--
+`userCert` defines the user certificate reference and is used for mTLS (you can ignore it when using one-way TLS)
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:1493]:
```suggestion
`userCert` defines the user certificate reference and is used for mTLS. When you use one-way TLS, you can ignore this property.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 19]
```diff
@@ -2218,6 +2510,36 @@ If the namespace is different, the config map or the secret is copied so that it
...
+
+
+[cols="1,1,1",options="header"]
+|===
+| Property | Type | Description
+
+| `additionalNamespaces`
+| `array (string)`
+| `additionalNamespaces` contains additional namespaces allowed to connect to the Network Observability namespace.
+It gives some flexibility in the network policy configuration, however should you need a more specific
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2534]:
```suggestion
It provides flexibility in the network policy configuration, but if you need a more specific
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 20]
```diff
@@ -2218,6 +2510,36 @@ If the namespace is different, the config map or the secret is copied so that it
...
+configuration, you can disable it and install your own instead.
+
+| `enable`
+| `boolean`
+| Set `enable` to `true` to deploy network policies on the namespaces used by Network Observability (main and privileged). It is disabled by default.
+These network policies better isolate the Network Observability components to prevent undesired connections to them.
+We recommend you either enable it, or create your own network policy for Network Observability.
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2541]:
```suggestion
Either enable it, or create your own network policy for Network Observability.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 21]
```diff
@@ -2375,6 +2697,12 @@ By convention, some values are forbidden. It must be greater than 1024 and diffe
 | `object`
 | scheduling controls how the pods are scheduled on nodes.
 
+| `secondaryNetworks`
+| `array`
+| Define secondary networks to be checked for resources identification.
+In order to guarantee a correct identification, it is important that the indexed values form an unique identifier across the cluster.
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2703]:
```suggestion
To guarantee a correct identification, the indexed values must form an unique identifier across the cluster.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 22]
```diff
@@ -2375,6 +2697,12 @@ By convention, some values are forbidden. It must be greater than 1024 and diffe
...
+If there are collisions in the indexes (same index used by several resources), those resources might be wrongly labeled.
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2704]:
```suggestion
If the same index is used by several resources, those resources might be wrongly labeled.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 23]
```diff
@@ -2440,6 +2768,52 @@ Type::
 
 
 
+== .spec.processor.advanced.secondaryNetworks
+Description::
++
+--
+Define secondary networks to be checked for resources identification.
+In order to guarantee a correct identification, it is important that the indexed values form an unique identifier across the cluster.
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2776]:
```suggestion
To guarantee a correct identification, the indexed values must form an unique identifier across the cluster.
```

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 24]
```diff
@@ -2440,6 +2768,52 @@ Type::
...
+If there are collisions in the indexes (same index used by several resources), those resources might be wrongly labeled.
```
Comment (lahinson) [inline modules/network-observability-flowcollector-api-specifications.adoc:2777]:
```suggestion
If the same index is used by several resources, those resources might be wrongly labeled.
```

**Review by lahinson — COMMENTED** [review]
Just a few suggestions for your consideration!

File: modules/network-observability-flowcollector-api-specifications.adoc (hunk 4 above)
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:215]: No it shouldn't

File: modules/network-observability-flowcollector-api-specifications.adoc (hunk 5 above)
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:218]: I also think that `+` shouldn't be there. Good point though about this one.

File: modules/network-observability-flowcollector-api-specifications.adoc (hunk 6 above)
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:417]: I think you're right here.

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 25]
```diff
@@ -115,6 +116,10 @@ Kafka can provide better scalability, resiliency, and high availability (for mor
 | `string`
 | Namespace where Network Observability pods are deployed.
 
+| `networkPolicy`
+| `object`
+| `networkPolicy` defines ingress network policy settings for Network Observability components isolation.
+
 | `processor`
 | `object`
 | `processor` defines the settings of the component that receives the flows from the agent,
```
[the remark this answers is not in the document]
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:125]: In my file, this comma precedes a continuation of the sentence: 

> `processor` defines the settings of the component that receives the flows from the agent,
_enriches them, generates metrics, and forwards them to the Loki persistence layer and/or any available exporter._

[the remark this answers is not in the document]
Comment (lahinson) [reply modules/network-observability-flowcollector-api-specifications.adoc:125]: Oh! That's cut off for me here. I should have expanded the file. Feel free to ignore my comment.

File: modules/network-observability-flowcollector-api-specifications.adoc (hunk 7 above)
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:464]: Yeah it should be eBPF I think.

File: modules/network-observability-flowcollector-api-specifications.adoc (hunk 8 above)
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:1290]: I think there's a space here so the period doesn't interfere with the link

File: modules/network-observability-flowcollector-api-specifications.adoc (hunk 22 above)
Comment (skrthomas) [reply modules/network-observability-flowcollector-api-specifications.adoc:2704]: I also think incorrectly is better than wrongly. Thanks for pointing this out.

**Comment by skrthomas** [comment]
Thanks for the review @lahinson !

File: modules/network-observability-flowcollector-api-specifications.adoc [hunk 26]
```diff
@@ -197,16 +202,21 @@ Otherwise it is matched as a case-sensitive string.
 
 | `features`
 | `array (string)`
-| List of additional features to enable. They are all disabled by default. Enabling additional features might have performance impacts. Possible values are: +
+| List of additional features to enable. They are all disabled by default. Enabling additional features might have performance impacts. A possible value is `+`.
```
Comment (jotak) [inline modules/network-observability-flowcollector-api-specifications.adoc:205]: hey @skrthomas , as commented in my PR, this change is incorrect

**Pull request merged**